src/decoders/router_decoder_custom_worker.erl

```diff
@@ -228,10 +280,16 @@ decode(
         )
     of
         {error, invalid_context} ->
-            Context1 = init_context(VM, Function),
+            {ok, Context1} = init_context(VM, Function),
             %% TODO: use Ferd's `backoff' library.
             %% See stateful use: ../device/router_device_channels_worker.erl
             decode(Payload, Port, UplinkDetails, State#state{context = Context1}, Retry - 1);
+        {error, Err} ->
+            %% TODO this eliminates any tolerance for intermittent
+            %% errors such as an occasional payload crashing their
+            %% JavaScript code, so maybe add `js_error_count` in
+            %% State and a threshold test here.
+            {{js_error, Err}, State#state{function = <<>>}};
```

Comment:
This one's tricky. The code didn't crash the VM, and is still valid because it was able to be loaded.
But every call after this type of failure will give them an error `ignoring_invalid_javascript`. 
It might be useful to check if the `Err` is a binary, then take up the first `\n`.
That would get shoved into the `decoded.error` key of what we send to their integration.

Reply:
ok. Let me add a test exercising the reporting of errors.  (Adding the extraction up to the first newline didn't crash any existing tests.)

Reply:
Addressed in 772469b, but I leave it to you for resolving this comment thread-- whether that commit is sufficient or not.